File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        res = 0
        for curr in range(len(height)):
            l = 0
            r = 0
            curr_bar = height[curr]
            for j in range(len(height)):
                if j < curr:
                    if height[j] > l:
                        l=height[j]
                else:
                    if height[j] > r:
                        r=height[j]
            if min(l,r) - height[curr] > 0:
                res += min(l,r) - height[curr]
        return res

File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int ans=0,curr;
        for(int i=0;i<heights.length;i++)
        {
            for(int j=0;j<heights.length;j++)
            {
                curr = heights[i];
                if(heights[i]>heights[j])
                {
                    curr = heights[j];
                }
                if(heights[i]<heights[j])
                {
                    curr = heights[i];
                }
                if(ans<curr*(j-i))
                {
                    ans = curr*(j-i);
                }
            }
        }
        return ans;
    }
}
